### PR TITLE
ESQL: Skip lookup fields when eliminating missing fields

### PR DESCRIPTION
**Pull request by alex-spies**
We do not have SearchStats for fields from lookup indices. And unfortunately, these are hard to obtain.

For now, just do not apply ReplaceMissingFieldWithNull to fields coming from an index used in `LOOKUP JOIN`. These are identified via their indexmode.

### CODE REVIEW

**Comment by alex-spies** [comment]
I contemplated adding also optimizer (unit) tests; not sure we want them, as those would have to test that a rule does nothing - which is not entirely correct, it just can't do anything at the moment due to missing SearchStats.

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec [hunk 1]
```diff
@@ -469,17 +469,101 @@ count:long | type:keyword
 3          | Error
 ;
 
-lookupWithFieldOnJoinKey-Ignore
```
Comment (alex-spies) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec:472]: This has become `filterOnJoinKeyAndRightSide`, one test case below.

**Review by costin — APPROVED** [review]
LGTM

**Comment by alex-spies** [comment]
CI only failed on [JdbcDocCsvSpecIT » test {docs.testFilterToday}](https://gradle-enterprise.elastic.co/s/fbzbeememcy5w/tests/:x-pack:plugin:sql:qa:server:single-node:javaRestTest/org.elasticsearch.xpack.sql.qa.single_node.JdbcDocCsvSpecIT/test%20%7Bdocs.testFilterToday%7D), which is in the [process of being fixed](https://github.com/elastic/elasticsearch/pull/118815) - so this should be fine to merge.

**Pull request merged**